servicepulse/saga-diagram.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Saga Diagram
+summary: The saga diagram illustrates how a saga is initiated, the messages it sends or handles, and details about saga data changes.
+component: ServicePulse
+reviewed: 2025-05-08
+related:
+- servicepulse/message-details
+---
+
+Sagas in NServiceBus coordinate long-running workflows by maintaining state, handling timeouts, and correlating related messages. ServicePulse offers the **Saga Diagram**, which renders a chronological visualization of state transitions and message flows for each saga instance. It provides a visual representation of saga execution, allowing users to trace how a saga instance processes incoming messages, changes state, and emits outgoing messages and timeouts. This visualization is especially useful for understanding complex workflows, debugging message flows, and validating long-running business processes.
+
+![Saga Diagram Overview](images/saga-diagram-overview.png 'width=800')
+
+## Prerequisites
+
+The Saga Diagram relies on audit data collected via the `NServiceBus.SagaAudit` plugin, which tracks saga state changes and message relationships, sending them to ServiceControl. ServicePulse uses this data to generate saga visualizations.
+
+To view saga diagrams in ServicePulse:
+
+1. Install the `NServiceBus.SagaAudit` package in all endpoints that are participating in the Saga:
+   ```
+   install-package NServiceBus.SagaAudit
+   ```
+
+2. Configure saga auditing in the endpoint configuration:
+   ```
+   endpointConfiguration.AuditSagaStateChanges(
+       serviceControlQueue: "particular.servicecontrol");
+   ```
+
+## How to Access the Saga Diagram
+
+The "Saga Diagram" tab in ServicePulse appears when viewing a message involved in a saga. If the message isn't part of a saga, the tab won't be visible.
+
+- Navigate to a message in ServicePulse (from the [All Messages](/servicepulse/all-messages.md) or [Failed Messages](/servicepulse/intro-failed-messages.md) view).
+- If the message is part of a saga, the "Saga Diagram" tab will be visible.
+- Click on the "Saga Diagram" tab to view the saga visualization.
+- The selected message is highlighted in blue in the "Saga Diagram".
+- If the saga audit plugin isn't installed, ServicePulse will display a notification with installation instructions.
+
+![Saga Diagram Plugin Needed](images/saga-diagram-plugin-needed.png 'width=400')
+
+## Saga Diagram Layout
+
+![Saga Diagram](images/saga-diagram.png 'width=800')
```

Comment:
Perhaps we can use an image of a Saga where the Saga Initiated update is not missing